We will work through the LAB for linear regression chapter in ISLR.

Basic functions that perform least squares linear regression and other simple analyses come
standard with the base distribution named MASS. So we use the library() function to run it.

In [58]:
lodalibraries=function(){
    library('MASS')
    library('ISLR')
    print('Libraries Loaded')
}

In [59]:
lodalibraries()

[1] "Libraries Loaded"


In [60]:
#1
fix(Auto)

In [61]:
attach(Auto)
slrf=lm(mpg~horsepower,data=Auto)
summary(slrf)


Call:
lm(formula = mpg ~ horsepower, data = Auto)

Residuals:
     Min       1Q   Median       3Q      Max 
-13.5710  -3.2592  -0.3435   2.7630  16.9240 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 39.935861   0.717499   55.66   <2e-16 ***
horsepower  -0.157845   0.006446  -24.49   <2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 4.906 on 390 degrees of freedom
Multiple R-squared:  0.6059,	Adjusted R-squared:  0.6049 
F-statistic: 599.7 on 1 and 390 DF,  p-value: < 2.2e-16


In [69]:
100*summary(slrf)$sigma/mean(mpg)
100*summary(slrf)$r.squared

[1] 20.92371

[1] 60.59483

1- Since the p-value is small then there is a relation between the response and predictor.


2- We use the value of RSE/mean of the response to have an idea on the error and how much it's big for this partucular problem 20,92% and R squared which is the amount of variability in mpg explained by horsepower is: 60.59%.

[1] 4.905757

[1] "call"          "terms"         "residuals"     "coefficients" 
 [5] "aliased"       "sigma"         "df"            "r.squared"    
 [9] "adj.r.squared" "fstatistic"    "cov.unscaled"